cirq-core/cirq/transformers/routing/mapping_manager.py

```diff
@@ -133,3 +136,12 @@ def shortest_path(self, lq1: 'cirq.Qid', lq2: 'cirq.Qid') -> Sequence['cirq.Qid'
     @cached_method
     def _physical_shortest_path(self, pq1: 'cirq.Qid', pq2: 'cirq.Qid') -> Sequence['cirq.Qid']:
         return nx.shortest_path(self._induced_subgraph, pq1, pq2)
+
+    def _value_equality_values_(self):
+        return self.graph_adjacency, self._map
```

Comment:
`_value_equality_values_` should return hashable objects but python dictionaries are not hashable by default. See https://github.com/quantumlib/Cirq/blob/23abcbeb0dbc90f4b15e889e943bf1a948232df3/cirq-core/cirq/devices/device.py#L136 for a way to return networkx graph data in a hashable format.

Reply:
I didn't realize these changes were included in #5829 but I also brought up the issue of hashy